src/runtime/BackingStore.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef __EscargotBackingStore__
+#define __EscargotBackingStore__
+
+namespace Escargot {
+
+using BackingStoreDeleterCallback = void (*)(void* data, size_t length,
+                                             void* deleterData);
+
+class BackingStore : public gc {
+    friend class ArrayBufferObject;
+
+public:
+    BackingStore(size_t byteLength);
+    BackingStore(void* data, size_t byteLength, BackingStoreDeleterCallback callback, void* callbackData);
+    ~BackingStore();
+
+    void* data() const
+    {
+        return m_data;
+    }
+
+    size_t byteLength() const
+    {
+        return m_byteLength;
+    }
+
+    bool isShared() const
+    {
+        return m_isShared;
+    }
+
+private:
+    // Indicates whether the backing store was created for an ArrayBuffer
+    bool m_isShared;
```

Comment:
if you named it for above purpose, It would be better to change the name to a more intuitive name and remove comment

Reply:
You are right!

Reply:
It is intended name. I want to use same variable name with v8

Reply:
Okay, I think it's an acceptable reason too